zebra/zebra_rib.c

```diff
@@ -457,9 +469,19 @@ nexthop_active_ipv4 (struct rib *rib, struct nexthop *nexthop, int set,
 	    {
 	      /* Directly point connected route. */
 	      newhop = match->nexthop;
-	      if (newhop && nexthop->type == NEXTHOP_TYPE_IPV4)
-		nexthop->ifindex = newhop->ifindex;
-	      
+	      if (newhop)
+		{
+		  if (nexthop->type == NEXTHOP_TYPE_IPV4)
+		    {
+		      nexthop->ifindex = newhop->ifindex;
+		      nexthop->type = NEXTHOP_TYPE_IPV4;
+		    }
+		  if (nexthop->type == NEXTHOP_TYPE_IPV6)
+		    {
+		      nexthop->ifindex = newhop->ifindex;
+		      nexthop->type = NEXTHOP_TYPE_IPV6_IFINDEX;
+		    }
+		}
```

Comment:
There was no change to `nexthop->type` in the previous code at all, in either function - why is it changed, and why is it different between IPv4 and IPv6?